utils/swift_build_support/tests/products/test_swift.py

```diff
@@ -497,3 +500,27 @@ def test_swift_tools_ld64_lto_codegen_only_for_supporting_targets(self):
              'TRUE'],
             [x for x in swift.cmake_options
                 if 'SWIFT_TOOLS_LD64_LTO_CODEGEN_ONLY_FOR_SUPPORTING_TARGETS' in x])
+
+    def test_swift_debuginfo_non_lto_args(self):
+        self.args.swift_debuginfo_non_lto_args = []
+        swift = Swift(
+            args=self.args,
+            toolchain=self.toolchain,
+            source_dir='/path/to/src',
+            build_dir='/path/to/build')
+        self.assertEqual(
+            ['-DSWIFT_DEBUGINFO_NON_LTO_ARGS:STRING='],
+            [x for x in swift.cmake_options
+                if 'SWIFT_DEBUGINFO_NON_LTO_ARGS' in x])
```

Comment:
This assertion is wrong and contradicts the help of `--swift-debuginfo-non-lto-args` -- if the flag is not provided (and/or set to `None`) I should not set the underlying CMake variable, so that the existing value can be pulled from the CMakeCache.txt file.